apps/take_picture/app.c

```diff
@@ -0,0 +1,152 @@
+/**
+ * This example takes a picture every 5s.
+   Example src: https://github.com/espressif/esp32-camera.git 
+ */
+
+
+
+
+// ================================ CODE ======================================
+
+#include <esp_event_loop.h>
+#include <esp_log.h>
+#include <esp_system.h>
+#include <nvs_flash.h>
+#include <sys/param.h>
+#include <string.h>
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include <rcl/rcl.h>
+#include <rcl/error_handling.h>
+#include <std_msgs/msg/int32.h>
+
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "boards.h"
+
+
+
+
+static const char *TAG = "example:take_picture";
+
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);vTaskDelete(NULL);}}
+#define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
+
+rcl_publisher_t publisher;
+std_msgs__msg__Int32 msg;
+camera_fb_t *pic;
+
+void timer_callback(rcl_timer_t * timer, int64_t last_call_time)
+{
+	UNUSED(last_call_time);
+	if (timer != NULL) {
+		RCSOFTCHECK(rcl_publish(&publisher, &msg, NULL));
+		// use pic->buf to access the image
+		msg.data = pic->len;
```

Comment:
Should we try to publish the image data on the topic? I'm not sure about which is the purpose of this demo code.

Reply:
My main purpose is to send the picture buffer, so it can be displayed on a computer. Here I sent the picture length as a proof of concept.